packages/cw3/src/query.rs

```diff
@@ -42,32 +50,63 @@ pub enum Cw3QueryMsg {
 }
 
 /// This defines the different ways tallies can happen.
-/// It can be extended as needed, but once the spec is frozen,
-/// these should not be modified. They are designed to be general.
+/// Every contract should support a subset of these, ideally all.
+///
+/// The total_weight used for calculating success as well as the weights of each
+/// individual voter used in tallying should be snapshotted at the beginning of
+/// the block at which the proposal starts (this is likely the responsibility of a
+/// correct cw4 implementation).
 #[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
 #[serde(rename_all = "snake_case")]
 pub enum ThresholdResponse {
-    /// Declares a total weight needed to pass
-    /// This usually implies that count_needed is stable, even if total_weight changes
-    /// eg. 3 of 5 multisig -> 3 of 6 multisig
+    /// Declares that a fixed weight of yes votes is needed to pass.
+    /// It does not matter how many no votes are cast, or how many do not vote,
+    /// as long as `weight` yes votes are cast.
+    ///
+    /// This is the simplest format and usually suitable for small multisigs of trusted parties,
+    /// like 3 of 5. (weight: 3, total_weight: 5)
+    ///
+    /// A proposal of this type can pass early as soon as the needed weight of yes votes has been cast.
     AbsoluteCount { weight: u64, total_weight: u64 },
-    /// Declares a percentage of the total weight needed to pass
-    /// This implies the percentage is stable, when total_weight changes
-    /// eg. at 50.1%, we go from needing 51/100 to needing 101/200
+
+    /// Declares a percentage of the total weight that must cast yes votes in order for
+    /// a proposal to pass. As with `AbsoluteCount`, it only matters the sum of yes votes.
+    ///
+    /// This is useful for similar circumstances as `AbsoluteCount`, where we have a relatively
+    /// small set of voters and participation is required. The advantage here is that if the
+    /// voting set (group) changes between proposals, the number of votes needed is adjusted
+    /// accordingly.
+    ///
+    /// Example: we set `percentage` to 51%. Proposal 1 starts when there is a `total_weight` of 5.
+    /// This will require 3 weight of yes votes in order to pass. Later, the Proposal 2 starts but the
+    /// `total_weight` of the group has increased to 9. That proposal will then automatically
+    /// require 5 yes of 9 to pass, rather than 3 yes of 9 as would be the case with `AbsoluteCount`.
```

Comment:
```suggestion
    /// Declares a percentage of the total weight that must cast Yes votes, in order for
    /// a proposal to pass. As with `AbsoluteCount`, Yes votes are the only ones that count.
    ///
    /// This is useful for similar circumstances as `AbsoluteCount`, where we have a relatively
    /// small set of voters, and participation is required.
    /// It is understood that if the voting set (group) changes between different proposals that refer to the same group, each proposal will work with a
    /// different set of voter weights (the ones snapshotted at proposal creation), and the passing weight for each proposal will be computed 
    /// depending on the absolute percentage times the total weights of the members at proposal creation.
    ///
    /// Example: we set `percentage` to 51%. Proposal 1 starts when there is a `total_weight` of 5.
    /// This will require 3 weight of yes votes in order to pass. Later, the Proposal 2 starts but the
    /// `total_weight` of the group has increased to 9. That proposal will then automatically
    /// require 5 yes of 9 to pass, rather than 3 yes of 9 as would be the case with `AbsoluteCount`.
```

Better?

Maybe the example is now redundant.

I'll still consider adding a paragraph about subtracting Abstain votes (if wee agree on that that should be considered here too).

Reply:
Looks good, and more explicit.

We say we only count Yes votes, so no need to mention Abstain explicitly, right?